File: backend/src/main/java/org/example/backend/service/UserService.java
package org.example.backend.service;

import lombok.RequiredArgsConstructor;
import org.example.backend.entity.User;
import org.example.backend.repository.UserRepository;
import org.springframework.stereotype.Component;
import reactor.core.publisher.Mono;

import java.util.List;

@Component
@RequiredArgsConstructor
public class UserService {
    private final UserRepository userRepository;
    public Mono<User> createUser(String name) {
        return userRepository.save(User.builder()
                        .name(name)
                .build());
    }

    public Mono<List<User>> getAllUsers() {
        return userRepository.findAll()
                .take(30)
                .collectList();
    }

    public Mono<User> getUserById(long id) {
        return userRepository.findById(id);
    }
}

File: shared/src/commonMain/kotlin/api/user/UserService.kt
package api.user

import model.GetUserResponse
import model.StoredUserResponse

interface UserService {
    suspend fun createUser(name: String): Result<Unit>

    suspend fun updateUser(): Result<Unit>

    suspend fun getAllUsers(): Result<StoredUserResponse>

    suspend fun getUserById(id: Long): Result<GetUserResponse>
}